### PR TITLE
upgrade batch relayer

### PR DESCRIPTION
**Pull request by groninge01**

### CODE REVIEW

File: modules/nav/NavbarPendingRewards.tsx [hunk 1]
```diff
@@ -44,7 +44,8 @@ export function NavbarPendingRewards() {
     } = useUserPendingRewards();
     const { stakedValueUSD, loading: userDataLoading } = useUserData();
     const { priceForAmount, getToken } = useGetTokens();
-    const loading = pendingRewardsLoading || userDataLoading;
+    const { data: hasMinterApproval, isLoading: isLoadingHasMinterApproval, refetch } = useHasMinterApproval();
+    const loading = pendingRewardsLoading || userDataLoading || isLoadingHasMinterApproval;
```
Comment (mrchickenburger) [inline modules/nav/NavbarPendingRewards.tsx:48]: nit: isLoading

File: modules/nav/NavbarPendingRewards.tsx [hunk 2]
```diff
@@ -59,18 +60,14 @@ export function NavbarPendingRewards() {
     const totalPendingRewardsUSD = pendingRewardsTotalUSD + pendingReliquaryRewardsTotalUSD;
 
     const { data: contractCalls } = useGaugeClaimGetContractCallData(
-        totalPendingRewardsUSD > 0.01,
-        pendingBALUSD > 0.01,
+        // totalPendingRewardsUSD > 0.01,
```
Comment (mrchickenburger) [inline modules/nav/NavbarPendingRewards.tsx:63]: Remove these if not needed anymore?

File: modules/nav/NavbarPendingRewards.tsx [hunk 3]
```diff
@@ -165,36 +162,31 @@ export function NavbarPendingRewards() {
...
+                                                buttonText="Approve BAL minting"
+                                            />
+                                        )}
+                                        {hasMinterApproval && (
                                             <BeetsSubmitTransactionButton
-                                                {...claimAllQuery}
-                                                isDisabled={pendingBALUSD < 0.01}
-                                                onClick={() => claimAllBAL()}
+                                                {...harvestQuery}
+                                                //isDisabled={pendingRewardsNonBALTotalUSD < 0.01 && pendingBALUSD < 0.01}
```
Comment (mrchickenburger) [inline modules/nav/NavbarPendingRewards.tsx:176]: Is this needed? Remove if not

File: modules/pool/stake/PoolUnstakeModal.tsx [hunk 4]
```diff
@@ -63,23 +61,14 @@ export function PoolUnstakeModal({ isOpen, onOpen, onClose }: Props) {
     const { pool } = usePool();
     const { withdraw, ...unstakeQuery } = useStakingWithdraw(pool.staking);
     const [steps, setSteps] = useState<TransactionStep[] | null>(null);
-    // TODO: fix comments below when relayer v6 is released
-    //const { data: hasMinterApproval, isLoading: isLoadingHasMinterApproval } = useHasMinterApproval();
-    //const loading = isLoadingBalances || isLoadingHasMinterApproval || isLoadingBatchRelayerApproval;
+    const { data: hasMinterApproval, isLoading: isLoadingHasMinterApproval } = useHasMinterApproval();
     const { data: hasBatchRelayerApproval, isLoading: isLoadingBatchRelayerApproval } = useHasBatchRelayerApproval();
-    const loading = isLoadingBalances || isLoadingBatchRelayerApproval;
+    const loading = isLoadingBalances || isLoadingHasMinterApproval || isLoadingBatchRelayerApproval;
```
Comment (mrchickenburger) [inline modules/pool/stake/PoolUnstakeModal.tsx:66]: nit: isLoading

**Review by mrchickenburger — APPROVED** [review]
Looks good, few cleanup comments but nothing major, since it's just getting rid of all the workarounds.

File: modules/nav/NavbarPendingRewards.tsx (hunk 2 above)
Comment (groninge01) [reply modules/nav/NavbarPendingRewards.tsx:63]: Good catch. Used that for testing with my shrimp rewards.

File: modules/nav/NavbarPendingRewards.tsx (hunk 3 above)
Comment (groninge01) [reply modules/nav/NavbarPendingRewards.tsx:176]: Same here.

**Pull request merged**